test/com/amazon/ion/impl/bin/WriteBufferTest.java

```diff
@@ -950,6 +950,58 @@ public void shiftBytesLeftAcrossBufferBlocks() throws IOException {
         assertBuffer("0123489ABCDEF".getBytes());
     }
 
+    @Test
+    public void shiftBytesLeftAcrossBufferBlocksEmptyingLastBlock() throws IOException {
+        assertEquals(11, ALLOCATOR.getBlockSize());
+        // The "B" is the first and only byte in the second block.
+        // Shifting 1 byte left by one empties the last block.
+        buf.writeBytes("0123456789AB".getBytes());
+        buf.shiftBytesLeft(1, 1);
+        assertBuffer("0123456789B".getBytes());
+    }
```

Comment:
Do/should we have tests for shifting bytes `==` to block size and `>=` block size?

Reply:
I can add tests for those. In practice, the largest shifts are by `2` and the block size is 32KB. However, at the unit test level there's no reason I can't/shouldn't test that.